app/api/speakers_calls.py

```diff
@@ -29,72 +31,89 @@ class Meta:
+    List and create Speakers Call
     """
 
     def query(self, view_kwargs):
-        query_ = self.session.query(CallForPaper)
-        if view_kwargs.get('id'):
-            query_ = query_.join(Event).filter(Event.id == view_kwargs['id'])
+        query_ = self.session.query(SpeakersCall)
+        if view_kwargs.get('event_id'):
+            query_ = query_.join(Event).filter(Event.id == view_kwargs['event_id'])
         elif view_kwargs.get('identifier'):
             query_ = query_.join(Event).filter(Event.identifier == view_kwargs['identifier'])
         return query_
 
     def before_create_object(self, data, view_kwargs):
-        if view_kwargs.get('id'):
-            event = self.session.query(Event).filter_by(id=view_kwargs['id']).one()
+        if view_kwargs.get('event_id'):
+            event = self.session.query(Event).filter_by(id=view_kwargs['event_id']).one()
             data['event_id'] = event.id
         elif view_kwargs.get('identifier'):
             event = self.session.query(Event).filter_by(identifier=view_kwargs['identifier']).one()
             data['event_id'] = event.id
 
     view_kwargs = True
     decorators = (jwt_required, )
-    schema = CallForPaperSchema
+    schema = SpeakersCallSchema
+    methods = ['POST', ]
     data_layer = {'session': db.session,
-                  'model': CallForPaper,
+                  'model': SpeakersCall,
                   'methods': {
                       'query': query,
                       'before_create_object': before_create_object
                   }}
 
 
-class CallForPaperDetail(ResourceDetail):
+class SpeakersCallDetail(ResourceDetail):
     """
-    cfs detail by id
+     speakers call detail by id
     """
     def query(self, view_kwargs):
-        query_ = self.session.query(CallForPaper)
-        if view_kwargs.get('id'):
-            query_ = query_.join(Event).filter(Event.id == view_kwargs['id'])
+        query_ = self.session.query(SpeakersCall)
+        if view_kwargs.get('event_id'):
+            query_ = query_.join(Event).filter(Event.id == view_kwargs['event_id'])
         elif view_kwargs.get('identifier'):
             query_ = query_.join(Event).filter(Event.identifier == view_kwargs['identifier'])
         return query_
 
+    def before_get_object(self, view_kwargs):
+        if view_kwargs.get('event_id'):
+            try:
+                event = self.session.query(Event).filter_by(id=view_kwargs['event_id']).one()
+            except NoResultFound:
+                raise ObjectNotFound({'parameter': 'event_id'},
+                                     "Event: {} not found".format(view_kwargs['event_id']))
+            else:
+                if event.speakers_call:
+                    view_kwargs['id'] = event.speakers_call.id
+                else:
+                    view_kwargs['id'] = None
+
+
     decorators = (jwt_required,)
-    schema = CallForPaperSchema
+    schema = SpeakersCallSchema
+    methods = ['GET', ]
```

Comment:
@niranjan94 suggested that this is a GET only method. Now I think it means, NOT POST( which anyways detail is, so we can remove this line altogether??) :P